### PR TITLE
App extensions mkting activities endpoint generator

### PR DESCRIPTION
**Pull request by kipply**
This pr adds generators for building app extensions (though only endpoints at the moment, no related graphql queries and such).

Specifically, this PR currently supports Marketing Activities, but is (hopefully) designed to support other extensions such as Shopify POS and Kit Skills (more info at https://help.shopify.com/en/api/embedded-apps/app-extensions)

### CODE REVIEW

File: marketing_activity_extension/generate-marketing-activity-extension.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  let lastImport;
+  let extensionRouterExists;
+  let hmacImportExists;
+
+  const hmacVerificationImport = `import verifyHmacRequest from "./api/hmac-verification.js";`;
+  const extensionImport = `import marketingActivitiesRouter from "./api/marketing-activities.js";`;
+  const routerCreation = `const apiRouter = new Router();`;
+  const routerConfiguration = `
+    apiRouter.prefix("/api");
+    apiRouter.use('/', async (ctx, next) => {
```
Comment (atesgoral) [inline marketing_activity_extension/generate-marketing-activity-extension.js]: Probably don't need the `'/'` path argument?

**Comment by atesgoral** [comment]
There's inconsistent usage of single/double quotes in generated code. We should standardize on one. I suggest single quotes.

File: marketing_activity_extension/generate-marketing-activity-extension.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  if (fs.existsSync(hmacFile)) {
+    process.exitCode = 2;
+  } else {
+    fs.writeFileSync(hmacFile, hmacVerificationTemplate, err => {
+      if (err) process.exitCode = 1;
+      console.log(`${hmacFile} was successfully created!`);
+    });
+  }
+
+
```
Comment (katiedavis) [inline marketing_activity_extension/generate-marketing-activity-extension.js]: might as well remove one of these

File: marketing_activity_extension/marketing-activities-template.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+const marketingActivitiesTemplate = `import Router from "koa-router";
+
+const handleResume = async ctx => {
+  ctx.body = {};
+  ctx.status = 202; // Accepted
+};
+const handlePause = async ctx => {
```
Comment (katiedavis) [inline marketing_activity_extension/marketing-activities-template.js]: can we add spaces between these for readability?

**Pull request merged**